lib/version_utils.pm

```diff
@@ -771,12 +771,13 @@ sub is_public_cloud {
 
 =head2 is_openstack
 
-Returns true if JEOS_OPENSTACK is set to 1
+Returns true if the tests loads Cloud image in OpenStack environment
+NO_CLOUD variable is set in order to test the image in QEMU
 
 =cut
 
 sub is_openstack {
-    return get_var('JEOS_OPENSTACK');
+    return !get_var('NO_CLOUD', '0') && get_var('FLAVOR', '') =~ /JeOS-for-OpenStack-Cloud.*/;
```

Comment:
```suggestion
    return get_var('FLAVOR', '') =~ /JeOS-for-OpenStack-Cloud.*/ && check_var('NO_CLOUD', '1');
```
Would switch the boolean arguments to go from the least specific to the most specific. Also would use `check_var` to avoid the double negation because of less required brain capacity to understand the logic.

Reply:
I agree with the check_var, but putting first the NO_CLOUD test which is more specific saves time in fact. Flavor is set everywhere and checking a simple boolean var (in case of get_var syntax) is simply fast

Reply:
I do not see the double negation though, it is only once there and both get_vars set a default variable

Reply:
I see the `!` and comparing it with `0` as double negation in the context of an `if` clause:

`!get_var('NO_CLOUD', '0') == check_var('NO_CLOUD', 1')`

I think that's better understandable yet both are correct.

Reply:
That is the thing, in order to have openstack environment you need to set NO_CLOUD=0 IOW the data source is OpenStack, not NoCloud. Hence, only at this point `is_openstack` returns true.
So it should be `check_var('NO_CLOUD', 0')` but I am unsure what check_var does when `NO_CLOUD` is undef.

Reply:
`check_var` only returns true if it matches, but false if it is undefined.

Reply:
I leave it up to you, the rest looks good!